Modules/Loadable/Segmentations/EditorEffects/qSlicerSegmentEditorPaintEffect.cxx

```diff
@@ -1416,39 +1402,32 @@ void qSlicerSegmentEditorPaintEffect::setupOptionsFrame()
 
   // Create options frame for this effect
   d->BrushDiameterFrame = new QFrame();
-  d->BrushDiameterFrame->setLayout(new QHBoxLayout());
+  QFormLayout* brushFormLayout = new QFormLayout();
+  brushFormLayout->setContentsMargins(0, 0, 0, 0);
+  d->BrushDiameterFrame->setLayout(brushFormLayout);
   paintOptionsLayout->addWidget(d->BrushDiameterFrame);
 
+  d->BrushDiameterSizeFrame = new QFrame();
+  d->BrushDiameterSizeFrame->setLayout(new QHBoxLayout());
+  paintOptionsLayout->addWidget(d->BrushDiameterSizeFrame);
+
   QLabel* diameterLabel = new QLabel(tr("Diameter:"), d->BrushDiameterFrame);
+  diameterLabel->setAlignment(Qt::AlignCenter);
   diameterLabel->setToolTip(tr("Set the paint brush size as percentage of screen size or as fixed length"));
-  d->BrushDiameterFrame->layout()->addWidget(diameterLabel);
 
-  d->BrushDiameterSpinBox = new qMRMLSpinBox(d->BrushDiameterFrame);
-  d->BrushDiameterSpinBox->setToolTip(tr("Set the paint brush size as percentage of screen size or as fixed length"));
-  d->BrushDiameterFrame->layout()->addWidget(d->BrushDiameterSpinBox);
-
-  QList<int> quickDiameters;
-  quickDiameters << 1 << 3 << 5 << 10 << 20 << 40;
-  foreach (int diameter, quickDiameters)
-    {
-    // QToolbutton is used instead of a regular push button to make the button smaller
-    QToolButton* quickDiameterButton = new QToolButton();
-    quickDiameterButton->setText(QString::number(diameter));
-    quickDiameterButton->setProperty("BrushDiameter", QVariant(diameter));
-    quickDiameterButton->setToolTip(tr("Set the paint brush size as percentage of screen size or as fixed length"));
-
-    d->BrushDiameterFrame->layout()->addWidget(quickDiameterButton);
-    QObject::connect(quickDiameterButton, SIGNAL(clicked()), d, SLOT(onQuickDiameterButtonClicked()));
-    }
-
-  d->BrushDiameterRelativeToggle = new QToolButton();
-  d->BrushDiameterRelativeToggle->setText("%");
+  d->BrushDiameterRelativeToggle = new QToolButton(d->BrushDiameterSizeFrame);
+  d->BrushDiameterRelativeToggle->setCheckable(true);
+  d->BrushDiameterRelativeToggle->setText("absolute");
+  d->BrushDiameterRelativeToggle->setChecked(false); // relative
```

Comment:
There is a contradiction here. If "Relative" button is set to "false" then its state is "absolute".
I would recommend to use the same naming convention as in markups module: `brushDiameterIsAbsoluteButton`. If it is checked then it means diameter is absolute (if unchecked then diameter is relative).